crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -467,27 +465,16 @@ pub fn handle_rename(world: ServerWorld, params: RenameParams) -> Result<Option<
         .into());
     }
 
-    let renames = world
+    let change = world
         .analysis()
         .rename(FilePosition { file_id, offset }, &*params.new_name)?;
-    if renames.is_empty() {
+    if change.is_none() {
```

Comment:
@matklad here checked if `change` is `None`

Reply:
Ah, indeed! It's more idiomatic to use 

```
let change = match change {
    None => return Ok(None),
    Some(it) => it,
}
```

That way, we have a type-system level gurantee that there will be no panics.